Add 1x1 specialization for conv with NCHW order

xiaomengy opened the pull request:
Summary: Add 1x1 specialization for conv with NCHW order

Differential Revision: D8944686

BIT-silence commented:
Some benchmark on CPU with N = 128, C = 128, H = 32, W = 32, M = 128, G = 4,
original impl: about 250ms
current impl: about 10ms

xiaomengy committed:
Add 1x1 specialization for conv with NCHW order (#9671)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/9671

Add 1x1 specialization for conv with NCHW order

Differential Revision: D8944686

fbshipit-source-id: 6a96813f3d9f0d47799391172214e3bf9d938f22

yinghai commented on caffe2/operators/batch_matmul_op.h:
```diff
@@ -255,7 +255,7 @@ class BatchMatMulOp final : public Operator<Context> {
 
       // TODO(T23893772): doing this in a loop is likely going to be slow on GPU
```
Does this comment still apply?

BIT-silence replied:
Actually I'm not sure about that. For the batmatmul op, we will update it later to also support broadcasting

yinghai reviewed (commented):
Could you provide some comment on what the difference is between `batchedgemm` and `batchedstridedgemm`?

houseroad reviewed (approved):
LGTM.

If using OSS, we can always do git clang-format on the staging files.
Mercurial may also support this.
